larpandoracontent/LArObjects/LArAdaBoostDecisionTree.cc

```diff
@@ -0,0 +1,299 @@
+/**
+ *  @file   larpandoracontent/LArObjects/LArAdaBoostDecisionTree.cc
+ *
+ *  @brief  Implementation of the lar adaptive boost decision tree class.
+ *
+ *  $Log: $
+ */
+
+#include "Helpers/XmlHelper.h"
+
+#include "larpandoracontent/LArObjects/LArAdaBoostDecisionTree.h"
+
+namespace lar_content
+{
+
+AdaBoostDecisionTree::AdaBoostDecisionTree() :
+    m_isInitialized(false)
+{
+    std::cout << "AdaBoostDecisionTree::AdaBoostDecisionTree" << std::endl;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+pandora::StatusCode AdaBoostDecisionTree::Initialize(const std::string &parameterLocation, const std::string &bdtName)
+{
+    if (m_isInitialized)
+    {
+        std::cout << "AdaBoostDecisionTree: bdt was already initialized" << std::endl;
+        return pandora::STATUS_CODE_FAILURE;
+    }
+
+    // ATTN: Add consistency check for number of features, bearing in min not all features in a bdt may be used
+    this->ReadXmlFile(parameterLocation, bdtName);
+ 
```

Comment:
Would probably be nicer to allocate the strong classifier here, rather than in the reading of the xml file. Need to check for success. Also need to police use of `m_pStrongClassifier` throughout, in case accidentally still used uninitialised (which should now be `nullptr` as per above).

Reply:
Done.